### PR TITLE
test: Fix testcase 84

### PR DESCRIPTION
**Pull request by gyuha-wa-vect0r**
Fix testcase 84 arg_mixed in ubuntu24.04

In 84 arg_mixed test, the mixed_sub is renamed to mixed_sub.specialized.1
in clang optimized builds so the test fails.
Fix not to optimize in mixed_sub function.

Start 1 tests without worker pool

Compiler                  gcc                                           clang                                       
Runtime test case         pg             finstrument-fu fpatchable-fun  pg             finstrument-fu fpatchable-fun
------------------------: O0 O1 O2 O3 Os O0 O1 O2 O3 Os O0 O1 O2 O3 Os  O0 O1 O2 O3 Os O0 O1 O2 O3 Os O0 O1 O2 O3 Os
084 arg_mixed           : OK OK OK OK OK SK SK SK SK SK OK OK OK OK OK  OK OK OK OK OK SK SK SK SK SK OK OK OK OK OK

### CODE REVIEW

**Review by honggyukim — COMMENTED** [review]
Thanks for the patch, but your description doesn't explain what the problem is.

I would rephrase the problem statement as follows.

The `mixed_sub` is renamed to `mixed_sub.specialized.1` in clang optimized builds so the test `84 arg_mixed` fails as follows.
```diff
$ ./runtest.py -vdp -O1 -c clang 084
        ...
t084_arg_mixed: diff result of clang -pg -O1 
--- expect      2024-09-01 22:24:43.922522624 +0900
+++ result      2024-09-01 22:24:43.922522624 +0900
@@ -2,3 +2,3 @@
    mixed_add(-1, 0.200000) = -0.800000;
-   mixed_sub(0x400000, 2048) = 0x3ff800;
+   mixed_sub.specialized.1();
    mixed_mul(-3.000000, 80000000000) = -240000000000;

084 arg_mixed           : NG
```
This can be fixed as follows.
```diff
diff --git a/tests/s-exp-mixed.c b/tests/s-exp-mixed.c
index 1e51173b..9f678937 100644
--- a/tests/s-exp-mixed.c
+++ b/tests/s-exp-mixed.c
@@ -7,6 +7,9 @@ double mixed_add(int a, float b)
        return a + b;
 }
 
+#if __clang__
+__attribute__((optnone))
+#endif
 long mixed_sub(void *a, unsigned long b)
 {
        return (long)a - b;
```
But I think it's not easy to pick which functions can be renamed due to clang optimization. @namhyung Do you have any better idea?

**Comment by namhyung** [comment]
I don't know if clang has an option to prevent the specialization.

**Comment by gyuha-wa-vect0r** [comment]
@honggyukim I fixed as you mentioned! Thank you!
